dcc-release-job/dcc-release-job-document/src/main/java/org/icgc/dcc/release/job/document/task/CreateVCFFileTask.java

```diff
@@ -74,29 +62,28 @@ public TaskType getType() {
   @Override
   @SneakyThrows
   public void execute(TaskContext taskContext) {
-    createVcfFiles(taskContext);
-    saveVcfFiles(taskContext);
-  }
+    val testedDonorCount = resolveTotalSsmTestedDonorCount(taskContext);
+    val releaseName = taskContext.getJobContext().getReleaseName();
+    val fastaFile = resolveFastaFile();
+
+    val header = getVCFHeaderRDD(testedDonorCount, releaseName, fastaFile, taskContext);
 
-  private void saveVcfFiles(TaskContext taskContext) throws IOException {
-    val fileSorter = new VCFFileSorter(tmpVcfFile, tmpVcfHeaderFile);
-    @Cleanup
-    val hdfsOutputStream = createOutputStream(taskContext);
-    fileSorter.sortAndSave(hdfsOutputStream);
+    val input = readDocInput(taskContext, FileType.MUTATION_CENTRIC_DOCUMENT);
+    val partitionsCount = getPartitionsCount(input);
+    val records = input.mapPartitions(new MutationVCFConverter(testedDonorCount, releaseName, properties))
+        .sortBy(key -> key, true, partitionsCount);
+
+    val output = header.union(records);
+    save(output, taskContext);
   }
 
-  private void createVcfFiles(TaskContext taskContext) throws IOException {
-    resolveTotalSsmTestedDonorCount(taskContext);
-    val mutationsReader = createMutationsReader(taskContext);
-    @Cleanup
-    val outputStream = createOutputStream();
-    @Cleanup
-    val writer = createMutationWriter(taskContext, resolveTotalSsmTestedDonorCount(taskContext), outputStream);
-
-    val mutationsIterator = mutationsReader.createMutationsIterator();
-    while (mutationsIterator.hasNext()) {
-      writer.write(mutationsIterator.next());
-    }
+  private void save(JavaRDD<String> output, TaskContext taskContext) {
+    val workingDir = taskContext.getJobContext().getWorkingDir();
+    val fileSystemSettings = Configurations.getSettings(taskContext.getFileSystem().getConf());
+
+    output.coalesce(1)
```

Comment:
Maybe add a comment as to why we are doing this and why it is 1. I know why, but the next guy might not be a Spark guru like you :)